ios/RNCSafeAreaView.m

```diff
@@ -47,16 +47,25 @@ - (NSString *)description
                                     NSStringFromUIEdgeInsets(_currentSafeAreaInsets)];
 }
 
-- (void)safeAreaInsetsDidChange
-{
-  [super safeAreaInsetsDidChange];
-  [self invalidateSafeAreaInsets];
-}
-
 - (void)didMoveToWindow
 {
+  UIView *previousProviderView = _providerView;
   _providerView = [self findNearestProvider];
+  BOOL providerViewDidChange = previousProviderView != _providerView;
+
   [self invalidateSafeAreaInsets];
+
+  if (providerViewDidChange) {
+    [NSNotificationCenter.defaultCenter
```

Comment:
Do we need to cleanup this observer when the view is removed from window too?

Reply:
Ok looks like it is not needed to remove explicitely in dealloc since it keeps weak refs (https://stackoverflow.com/a/36822456). We can keep this as is.